Automattic/wp-calypso
# Update to Node 10 (10.4.0) and npm 6.1.0

As an alternative to moving to just npm@6 in #25002, move up to the latest in the next LTS line, 10.4.0,. which included npm@6.1.0. 

~Holding off on adding the "needs e2e" label until we can get the basic test suite to pass.~ They passed! Added e2e tests. (@blowery)

## Testing Instructions
This is similar to most other node upgrades. See https://github.com/Automattic/wp-calypso/pull/25004, https://github.com/Automattic/wp-calypso/pull/23744, et all

### Local Dev
- Switch to node version 10.4.0 (npm 6.1.0) (an `nvm i` in the root will do it if you use `nvm`)
- `npm run distclean`
- `npm start`
- No functional changes to Calypso

### Local Docker Image
- Follow image instructions at: PCYsg-5XR-p2 or
- Setup docker locally and add the following host entry:
  ```
  127.0.0.1 wpcalypso.wordpress.com
  ```
- `npm run build-docker`
- `npm run docker`

### Desktop
- Clone the wp-desktop project
- Follow install instructions
- In the calypso submodule checkout this branch
- Desktop behaves normally

Desktop companion PR: https://github.com/Automattic/wp-desktop/pull/453
